web/modules/custom/brms_sor/src/Service/FmsSorApi.php

```diff
@@ -0,0 +1,252 @@
+      }
+    }
+
+    try {
+      $response = $this->httpClient->request('POST', $this->authUrl, [
+        'headers' => [
+          'Content-Type' => 'application/x-www-form-urlencoded',
+        ],
+        'form_params' => [
+          'grant_type' => 'client_credentials',
+          'client_id' => $this->clientId,
+          'client_secret' => $this->clientSecret,
+        ],
+      ]);
+
+      $data = $this->json->decode($response->getBody()->getContents());
+      $this->cache->set('brms_sor_fms_access_token', $data['access_token'], $this->time->getRequestTime() + $data['expires_in']);
+
+      return $data['access_token'];
+    }
+    catch (\Exception $e) {
+      $this->logger->get('brms_sor')->error('Failed to get access token from the FMS API: @message', ['@message' => $e->getMessage()]);
+      throw $e;
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function get(string $call, array $data = [], bool $use_cached_token = TRUE): array {
+    try {
+      $token = $this->getAccessToken($use_cached_token);
+      $headers = [
+        'Authorization' => 'Bearer ' . $token,
+      ];
+
+      $args = ['headers' => $headers, 'query' => $data];
+      $uri = $this->baseUrl . $call;
+      $this->response = $this->httpClient->get($uri, $args);
+      return $this->json->decode($this->response->getBody()->getContents());
+    }
+    catch (RequestException $e) {
+      $this->response = $e->getResponse();
+      $this->logger->get('brms_sor')->error('Failed to call GET @uri on FMS API: @message',
+        ['@uri' => $uri, '@message' => $e->getMessage()]);
+      // If the token expired, try again with a new token.
+      if ($e->getCode() === 401 && $use_cached_token) {
+        return $this->get($call, $data, FALSE);
+      }
+      else {
+        throw $e;
+      }
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function patch(string $call, array $data = [], bool $use_cached_token = TRUE): array {
+    try {
+      $token = $this->getAccessToken($use_cached_token);
+      $headers = [
+        'Authorization' => 'Bearer ' . $token,
+        'Content-type' => 'application/vnd.api+json',
+      ];
+
+      $args = ['headers' => $headers, 'json' => $data];
+      $uri = $this->baseUrl . $call;
+      $this->response = $this->httpClient->patch($uri, $args);
+      return $this->json->decode($this->response->getBody()->getContents());
+    }
+    catch (RequestException $e) {
+      $this->response = $e->getResponse();
+      $this->logger->get('brms_sor')->error('Failed to call PATCH @uri on FMS API: @message',
+        ['@uri' => $uri, '@message' => $e->getMessage()]);
+      // If the token expired, try again with a new token.
+      if ($e->getCode() === 401 && $use_cached_token) {
+        return $this->patch($call, $data, FALSE);
+      }
+      else {
+        throw $e;
+      }
+    }
+  }
+
+  /**
+   * Get the UUID of a CFR by its global ID.
+   */
+  public function getCfrUuid(string $cfr_global_id): ?string {
```

Comment:
If the method is public, should be defined in the interface and the comment signature should be there.
If it's an internal method, it should be protected/private.

Reply:
added method to the interface